network/src/protocols/discovery/mod.rs

```diff
@@ -422,28 +419,26 @@ fn create_note(
     peer_info: PeerInfo,
     full_node_payload: FullNodePayload,
 ) -> Note {
-    let peer_info_bytes = peer_info
-        .write_to_bytes()
-        .expect("Protobuf serialization fails");
+    let peer_info_bytes = peer_info.to_bytes().expect("Protobuf serialization fails");
     let peer_info_signature = sign(&signer, &peer_info_bytes);
 
-    let mut signed_peer_info = SignedPeerInfo::new();
-    signed_peer_info.set_peer_info(peer_info_bytes.into());
-    signed_peer_info.set_signature(peer_info_signature.into());
+    let mut signed_peer_info = SignedPeerInfo::default();
+    signed_peer_info.peer_info = peer_info_bytes.to_vec();
+    signed_peer_info.signature = peer_info_signature;
 
     let payload_bytes = full_node_payload
-        .write_to_bytes()
+        .to_bytes()
         .expect("Protobuf serialization fails");
     let payload_signature = sign(&signer, &payload_bytes);
 
-    let mut signed_full_node_payload = SignedFullNodePayload::new();
-    signed_full_node_payload.set_payload(payload_bytes.into());
-    signed_full_node_payload.set_signature(payload_signature.into());
+    let mut signed_full_node_payload = SignedFullNodePayload::default();
+    signed_full_node_payload.payload = payload_bytes.to_vec();
+    signed_full_node_payload.signature = payload_signature;
 
-    let mut note = Note::new();
-    note.set_peer_id(peer_id.into());
-    note.set_signed_peer_info(signed_peer_info);
-    note.set_signed_full_node_payload(signed_full_node_payload);
+    let mut note = Note::default();
+    note.peer_id = peer_id.into();
+    note.signed_peer_info = Some(signed_peer_info);
```

Comment:
Does prost automatically assume Option for sub-message types?

Reply:
That's by design in proto3 apparently

Reply:
https://stackoverflow.com/questions/42622015/how-to-define-an-optional-field-in-protobuf-3/42634681#42634681